pylint/checkers/variables.py

```diff
@@ -722,28 +723,51 @@ def _uncertain_nodes_in_except_blocks(
                 if (
                     isinstance(node_statement.parent, nodes.TryFinally)
                     and node_statement in node_statement.parent.finalbody
-                    # We have already tested that other_node_statement has two parents
-                    # and it was TryExcept, so getting one more parent is safe.
-                    and other_node_statement.parent.parent.parent.parent_of(
-                        node_statement
-                    )
+                    and closest_try_except.parent.parent_of(node_statement)
                 ):
                     uncertain_nodes.append(other_node)
-                else:
-                    # Assume the except blocks execute. Possibility for a false negative
-                    # if one of the except blocks does not define the name in question,
-                    # raise, or return. See: https://github.com/PyCQA/pylint/issues/5524.
+                # Assume the except blocks execute, so long as each handler
+                # defines the name, raises, or returns.
+                elif all(
+                    NamesConsumer._defines_name_raises_or_returns(node.name, handler)
+                    for handler in closest_try_except.handlers
+                ):
                     continue
 
-            if NamesConsumer._check_loop_finishes_via_except(
-                node, other_node_statement.parent.parent
-            ):
+            if NamesConsumer._check_loop_finishes_via_except(node, closest_try_except):
                 continue
 
             # Passed all tests for uncertain execution
             uncertain_nodes.append(other_node)
         return uncertain_nodes
 
+    @staticmethod
+    def _defines_name_raises_or_returns(
+        name: str, handler: nodes.ExceptHandler
+    ) -> bool:
+        """Return True if some child of `handler` defines the name `name`,
+        raises, or returns.
+        """
+        for stmt in handler.get_children():
+            if isinstance(stmt, (nodes.Raise, nodes.Return)):
+                return True
+            if isinstance(stmt, nodes.Assign) and stmt.targets[0].as_string() == name:
```

Comment:
```suggestion
            if isinstance(stmt, nodes.Assign):
                 for target in stmt.targets:
                     if isinstance(target, nodes.AssignName) and target.name == name:
                         return True
                     if isinstance(target, (nodes.Tuple, nodes.List)):
                         for elt in utils.get_all_elements(target):
                             if isinstance(elt, nodes.Const) and elt.value == name:
                                 return True
```

Three parts to this suggestion. 

1) I figured out the case where `len(stmt.targets) > 1`. Looping over `stmt.targets` catches this case:

```
except ZeroDivisionError:
   bsg = msg = "Division by 0"
print(msg)
``` 

2) `as_string` is primarily used to format message parameters, and it seems better to avoid it in comparisons. 

3) Check target unpacking to `List` and `Tuple`, which can recursively contain more `List` or `Tuple` to catch this case:

```
asd, msg = ("asd", "Division by 0")
[asd, msg] = ["asd", "Division by 0"]
```

Reply:
Thank you, this is fantastic help.